File: Lab3Csharp/Lab3Csharp/SingleDimensionArray.cs
using System.Linq;

namespace Lab3Csharp
{
    internal class SingleDimensionArray<E>
    {
        public SingleDimensionArray(E element) : this(new E[] { element })
        {
        }

        public SingleDimensionArray(E[] elements)
        {
            Elements = elements ?? throw new System.ArgumentNullException("Elements are null");
        }

        public E[] Elements { get => Elements; set => Elements = value; }

        public E this[int index]
        {
            get => Elements[index];
            set => Elements[index] = value;
        }

        public E GetMax() => Elements.Max();

        public E GetMin() => Elements.Min();

        public override string ToString() => string.Join("\t", Elements.Select(e => e.ToString()).ToArray());
    }
}

File: Lab3Java/src/SingleDimensionArray.java
import java.util.Arrays;

public class SingleDimensionArray {
    private double[] elements;

    SingleDimensionArray(double[] elements) {
        if (elements == null)
            throw new NullPointerException("Array refers to null");
        else this.elements = elements;
    }

    double getElement(int index) {
        if (index < -1 || index > elements.length)
            throw new ArrayIndexOutOfBoundsException("Index: " + index +
                    "array length: " + elements.length);
        return elements[index];
    }

    void setElement(int index, double value) {
        if (index < -1 || index > elements.length)
            throw new ArrayIndexOutOfBoundsException("Index: " + index +
                    "array length: " + elements.length);
        elements[index] = value;
    }

    double getMax() {
        double max = elements[0];
        for (int i = 1; i < elements.length; i++) {
            if (elements[i] > max)
                max = elements[i];
        }
        return max;
    }

    double getMin() {
        double min = elements[0];
        for (int i = 1; i < elements.length; i++) {
            if (elements[i] < min)
                min = elements[i];
        }
        return min;
    }

    @Override
    public String toString() {
        return "Array elements={" + Arrays.toString(elements) +
                '}';
    }
}
